src/TypeSpec.Extension/Emitter.Csharp/src/lib/client-model-builder.ts

```diff
@@ -85,12 +85,30 @@ export function createModelForService(
 
     const apiVersions: Set<string> | undefined = new Set<string>();
     let defaultApiVersion: string | undefined = undefined;
-    const versions = getVersions(program, service.type)[1]?.getVersions();
+    let versions = getVersions(program, service.type)[1]
+        ?.getVersions()
+        .map((v) => v.value);
+    if (sdkContext.emitContext.options["api-version"] !== undefined) {
```

Comment:
Need consider the scenario "api-version" = "all" where it should contain all the api-version enum value

Reply:
I remember Chenjie told me not to try `all` which is still pending.